IdentityCore/src/cache/metadata/accountMetadata/MSIDAccountMetadataCacheItem.m

```diff
@@ -113,6 +127,23 @@ - (NSDictionary *)jsonDictionary
     return dictionary;
 }
 
+- (NSString *)URLMapSubkey:(BOOL)instanceAware
+{
+    // The subkey is in the of format of @"URLMap-key1=value1&key2=value2...",
+    // where key1, key2... are the request parameters that may affect the url mapping.
+    // Currently the only parameter that affects the mapping is instance aware flag.
+    //
+    // Example of subkeys:
+    // "URLMap-" : with all possible keys being their default value repectively. Default
+    //             value of instance_aware is NO, so "URLMap-" represents "URLMap-instance_aware=NO"
+    // "URLMap-instance_aware=YES" : with instance_aware being YES.
+    //
+    // The benefit of such a design is, if we are introducing new parameters what will affect the
+    // mapping, there will be no breaking change to existing clients who don't use the new parameters.
+    
+    return instanceAware ? @"URLMap-instance_aware=YES" : @"URLMap-";
```

Comment:
nit: do we need double structure? It will look a bit funny :)
{
"urlmap": {"urlmap": ""}
}

Reply:
+1, since "URLMap" is part of the key, we could use flat dictionary here.

Reply:
+1, addressed :)